examples/crds/ciliumegressnatpolicies.yaml

```diff
@@ -0,0 +1,190 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.3.1-0.20200716001835-4a903ddb7005
+  creationTimestamp: null
+  name: ciliumegressnatpolicies.cilium.io
+spec:
+  group: cilium.io
+  names:
+    kind: CiliumEgressNATPolicy
+    listKind: CiliumEgressNATPolicyList
+    plural: ciliumegressnatpolicies
+    singular: ciliumegressnatpolicy
+  scope: Cluster
+  versions:
+  - additionalPrinterColumns:
+    - description: The age of the identity
+      jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v2
```

Comment:
Since this is a new feature and may end up getting used by more than one provider, and may eventually end up being a resource type in K8s itself (or something like it), we need to change this from v2 to "v2alpha1". See [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/) for details.

Reply:
Ack. Thanks. Is that something we need to cover in this PR?

Reply:
Yes.

Reply:
+1. I imagine to move new go structs from `v2/types.go` to `v2alpha1/types.go`, autogen should take care of the rest.

Reply:
Finally figured this out. It's updated in the new iteration.